tests/test_gettemplates.F90

```diff
@@ -0,0 +1,502 @@
+    & char(1), char(0), char(0), char(0), char(0), char(0),    &
+    & char(3), char(0), char(0), char(0), char(n2), char(7),   & 
+    & char(n1), char(11), char(15), char(12), char(20),         &
+    & char(10), char(1), char(0), char(0), char(0), char(0),   &
+    & char(1), char(0), char(1), char(0), char(0), char(0),    &
+    & char(12), char(1), char(0), char(0), char(0), char(0),   &
+      ! section 5
+    & char(0), char(0), char(0), char(47), char(5),  char(0),  &
+    & char(0), char(0), char(k10), char(0), char(2),  char(0), &
+    & char(0), char(0), char(0), char(0), char(0),  char(0),   &
+    & char(1), char(8), char(1), char(1), char(1), char(Nan),  &
+    & char(Nan), char(Nan), char(Nan), char(0), char(0),        &
+    & char(0), char(0), char(0), char(0), char(0), char(0),    &
+    & char(0), char(0), char(0), char(0), char(0), char(0),    &
+    & char(0), char(0), char(0), char(0), char(0), char(0),    &
+      ! section 6
+    & char(0), char(0), char(0), char(6), char(6),  char(n3),  &
+      ! section 7
+    & "7", "7", "7", "7", char(0), char(0), &
+    & char(0), char(12), char(7),  char(1), char(0), char(0), &
+      ! section 8
+    &  char(0), char(0), char(0),  char(0) /) 
+
+  ! Grib message bad section 3 
+  fgrib_3(:)=(/                                                    &
+  ! section 0
+  & "G", "R", "I", "B", char(0), char(0),                        &
+  & char(0), char(2), char(0), char(0), char(0), char(0),    &
+  & char(0), char(0), char(1), char(13),                       &
+    ! section 1
+  & char(0), char(0), char(0), char(21), char(1), char(0),   &
+  & char(8), char(0), char(0), char(1), char(0), char(1),    &
+  & char(7), char(n1), char(11), char(15), char(10),          &
+  & char(10), char(10), char(0), char(1),                      &
+    ! section 2
+  & char(0), char(0), char(0), char(11), char(2),             &
+  & char(1), "H", "K", "T", "A", "R",                             &
+    ! section 3
+  & char(0), char(0), char(0), char(81), char(3), char(0),   &
+  & char(0), char(0), char(0), char(k10), char(0), char(0),  &
+  & char(0), char(30), char(1), char(0), char(k1), char(k1), &
+  & char(k1), char(k1), char(0), char(0), char(0), char(0),  &
+  & char(0), char(0), char(0), char(0), char(0), char(0),    &
+  & char(k2), char(k2), char(k2), char(k2), char(k3),         &
+  & char(k3), char(k3), char(k3), char(k4), char(k4),         &
+  & char(k4), char(k4),char(k5),char(k5), char(k5),           &
+  & char(k5), char(0), char(k6), char(k6), char(k6),          &
+  & char(k6), char(k7), char(k7), char(k7), char(k7),         &
+  & char(k8), char(k8), char(k8), char(k8), char(k8),         &
+  & char(k8), char(k8), char(k8), char(0), char(80),          &
+  & char(k6), char(k6), char(k6), char(k6), char(k6),         &
+  & char(k6), char(k6), char(k6), char(k9), char(k9),         &
+  & char(k9), char(k9), char(0), char(0), char(0), char(0),  &
```

Comment:
This is fine but take a look in test_gribcreate.F90 and you'll see a different way.

Only store one valid GRIB2 message, but then use code like this to change a value and run a test:

  ! Change the first byte of the message, then try to add grid - will
  ! not work.
  old_val = cgrib(1)
  cgrib(1) = char(0)
  call addgrid(cgrib, lcgrib, igds, igdstmpl, igdstmplen, &
       ideflist, idefnum, ierr)
  if (ierr .ne. 1) stop 50
  cgrib(1) = old_val

Note that I change the first byte, then try an addgrid() call, and it fails. THen I reset the first byte to its correct value, ready for the next test.

Using this technique, you should be able to reach most of the other error checking code...

Reply:
???